_drafts/2018-06-13-upgrade-sensenet-6-to-7.md

```diff
@@ -0,0 +1,195 @@
+
+### Available in a compatibility pack
+The [compatibility pack](https://github.com/SenseNet/sn-compatibility-pack) is an API package that will be installed during the upgrade. It contains many retired APIs that will bring at least parts of removed features back. Sometimes they only let you access your data (e.g. tags), in other cases you will still be able to use the full feature.
+
+- Wall
+- Journal
+- Survey
+- Voting
+- Rating
+- Wiki
+- Blog
+- Discussion forum
+- Event calendar
+- Tag management UI
+
+The compatibility pack is not maintained anymore in sensenet 7 (of course the community may fix bugs in it or add missing APIs), but at least it lets upgraded sites operate without too much change. It is recommended to replace these features with a more modern implementation before or after the upgrade.
+
+## Components and future patches
+If you upgrade a sensenet 6 instance to sensenet 7, you will have a 'regular' sensenet 7 installation. You will be able to install **all future patches** the same way as any other sensenet 7 instance. There are a few things to consider here though:
+
+- You'll have all the currently available sensenet 7 components, you **cannot opt-out** of any of them during patching (and currently there is no way to uninstall a component later).
+- Your repository will contain lots of old or obsolete content that will not be deleted during patching. They will cause no harm, but it is advisable to make a cleanup before upgrading - meaning deleting demo or unnecessary content.
+
+## sensenet 7 - a different philosophy
+sensenet 6 was a monolithic application that gave developers the possibility to **extend it** with their custom code and features - but even without that it provided many features out of the box, including a full UI.
+
+sensenet 7 however is a platform that you can **integrate into your application** (notice the difference?). This gives developers a lot more flexibility and front-end developers a lot more options to build a UI using modern technologies.
+
+This is why working with sensenet 7 is different in many ways from what you are used to with sensenet 6. For example:
+
+- we provide libraries and installers as **NuGet and npm packages** (instead of dlls to copy as in sensenet 6)
+- our components are versioned and upgraded separately and more frequently as fixes and features are published
+- it is a lot easier to participate in the dev process by monitoring our repositories on [GitHub](https://github.com/SenseNet) and contributing to our source code
+
+### A new UI development approach 
+In sensenet 7 projects the UI is created using modern technologies (like a one-page application with React or ASP.NET MVC) instead of ASP.NET *WebForms* (and portlets and ascx controls) which is still available as a legacy technology but *should not be used in new projects* - please use one of our many [front-end libraries](/blog/2018/02/21/scoped-packages) that let you compose UI elements in a flexible way.
```

Comment:
```suggestion
In sensenet 7 projects the UI is created using modern technologies (like a single-page application with React or ASP.NET MVC) instead of ASP.NET *WebForms* (and portlets and ascx controls) which is still available as a legacy technology but *should not be used in new projects* - please use one of our many [front-end libraries](/blog/2018/02/21/scoped-packages) that let you compose UI elements in a flexible way.
```